### DISCUSSION TITLE
Pagination component (willing to work on a PR)

### DISCUSSION THREAD

**Discussion by AnisTigrini** [chakra-ui/chakra-ui#8417] [Ideas]
### Description

Hi there Chackra team, hope you are all well. I think I saw this same issue pop up a few times, but a pagination component would save a lot of time to all the users. Having a component that is similar to what Material UI offers would be amazing. I would be willing to work on a PR for this :).

### Problem Statement/Justification

Right now there is no Pagination component and users have to create it from scratch.
Since this is a pretty redundant task and behavior of the component is suited to be centralized in a component, 
this would be a useful feature for all the users of Chackra UI.

### Proposed Solution or API

The API that I would propose would be similar to what Material UI has implemented.
A component with the following props:
- Current page
- Amount of pages
- Boundary count (number of visible pages at the beginning and the end).
- Right and left Icon 

### Alternatives

There are no current alternatives that I would know of at the moment (other than creating it from scratch).

### Additional Information

Materiaul UI pagination component : https://mui.com/material-ui/react-pagination/

**Comment by AnisTigrini** [comment]
Let me now if this was in your plans @segunadebayo.
I did see some interest in such a feature. 
I would be able to go forward and start working on it if I have the green light.

Let me know,
Thanks,

**Comment by veloware** [comment]
Someone may be along with the official answer, but Ark has pagination components, so it looks like it will either be in Chakra v3 or at least you can roll your own with Ark - https://ark-ui.com/docs/components/pagination

**Comment by AnisTigrini** [comment]
Well I was wondering if there was a need to add it to the core components of Chackra instead of relying on other libraries.
Let me know for anything 👍 .

**Comment by niklas-paro** [comment]
Looks like there is a Pagination component in Chakra v3, using Ark under the hood:
https://github.com/chakra-ui/chakra-ui/blob/da0a70c8e71a1aac8d7dda6b3cd2f08568d99bd6/packages/react/src/components/pagination/pagination.tsx

**Comment by segunadebayo** [comment]
`Pagination` shipped. https://chakra-ui.com/docs/components/pagination
